packages/rhi-webgl/src/GLTexture2DArray.ts

```diff
@@ -0,0 +1,111 @@
+
+    if (!this._isWebGL2) {
+      throw new Error(`Texture2D Array is not supported in WebGL1.0`);
+    }
+
+    if (!GLTexture._supportTextureFormat(format, rhi)) {
+      throw new Error(`Texture format is not supported:${TextureFormat[format]}`);
+    }
+
+    this._formatDetail = GLTexture._getFormatDetail(format, this._gl, true);
+    this._gl.texStorage3D(this._target, mipmapCount, this._formatDetail.internalFormat, width, height, length);
+  }
+
+  /**
+   * {@inheritDoc IPlatformTexture2DArray.setPixelBuffer}
+   */
+  setPixelBuffer(
+    elementIndex: number,
+    colorBuffer: ArrayBufferView,
+    mipLevel: number = 0,
+    x: number = 0,
+    y: number = 0,
+    width?: number,
+    height?: number
+  ): void {
+    const { _target: target, _gl: gl } = this;
+    const { internalFormat, baseFormat, dataType, isCompressed } = this._formatDetail;
+
+    width = width || Math.max(1, this._texture.width >> mipLevel) - x;
+    height = height || Math.max(1, this._texture.height >> mipLevel) - y;
+
+    this._bind();
+    gl.pixelStorei(gl.UNPACK_FLIP_Y_WEBGL, 0);
+    gl.pixelStorei(gl.UNPACK_PREMULTIPLY_ALPHA_WEBGL, 0);
+
+    if (isCompressed) {
+      gl.compressedTexSubImage3D(target, mipLevel, x, y, elementIndex, width, height, 1, internalFormat, colorBuffer);
+    } else {
+      gl.texSubImage3D(target, mipLevel, x, y, elementIndex, width, height, 1, baseFormat, dataType, colorBuffer);
+    }
+  }
+
+  /**
+   * {@inheritDoc IPlatformTexture2DArray.setImageSource}
+   */
+  setImageSource(
+    elementIndex: number,
+    imageSource: TexImageSource,
+    mipLevel: number,
+    flipY: boolean,
+    premultiplyAlpha: boolean,
+    x: number,
+    y: number
+  ): void {
+    const gl = this._gl;
+    const { baseFormat, dataType } = this._formatDetail;
+
+    this._bind();
+    gl.pixelStorei(gl.UNPACK_FLIP_Y_WEBGL, +flipY);
+    gl.pixelStorei(gl.UNPACK_PREMULTIPLY_ALPHA_WEBGL, +premultiplyAlpha);
+    gl.texSubImage3D(
+      this._target,
+      mipLevel,
+      x,
+      y,
+      elementIndex,
+      imageSource.width,
+      imageSource.height,
+      1,
+      baseFormat,
+      dataType,
+      imageSource
+    );
+  }
+
+  /**
+   * {@inheritDoc IPlatformTexture2DArray.getPixelBuffer}
+   */
+  getPixelBuffer(
+    elementIndex: number,
+    x: number,
+    y: number,
+    width: number,
+    height: number,
+    mipLevel: number,
+    out: ArrayBufferView
+  ): void {
+    const { _gl: gl, _formatDetail: formatDetail } = this;
+
+    if (formatDetail.isCompressed) {
+      throw new Error("Unable to read compressed texture");
+    }
+
+    gl.bindFramebuffer(gl.FRAMEBUFFER, this._getReadFrameBuffer());
+    gl.framebufferTextureLayer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, this._glTexture, mipLevel, elementIndex);
+    gl.readPixels(x, y, width, height, formatDetail.baseFormat, formatDetail.dataType, out);
+    gl.bindFramebuffer(gl.FRAMEBUFFER, null);
+  }
```

Comment:
why not refactor `GLTexture._getPixelBuffer` ?

Reply:
Will cause _getPixelBuffer to be too complicated, `Texture2DArray` uses framebufferTextureLayer, there is no reuse code